### ISSUE TITLE
Greyed out header...

### ISSUE THREAD

**Issue by JonathanMagus** [fontforge/fontforge#2626]
I am not sure if this is a bug, or something that Fontforge is indicating that I don't understand:

In a font that I am making (a digital re-mastering of an historic typeface), which passes validation correctly without error on saving, the header cell of the apostrophe (8217, quoteright) glyph, as seen in the main view of the interface (Unicode points/characters in a grid), remains grey when saved, whereas all other cell heads revert to white.

Nothing I have tried makes any difference to this and no other glyph is affected.

There is no hinting, deliberately, in this font, however, blue values are in place.

The font works well. The consistent inconsistency of how this character presents in the interface is worrying though, since it suggests something is wrong.

Is this a bug, or is there another explanation?

See attached:

![apostrophe](https://cloud.githubusercontent.com/assets/8567632/13088305/a4b81390-d4e3-11e5-9772-92e4e2358f15.png)

**Comment by frank-trampe** [comment]
Does this happen on your installation with other typefaces? Which typeface are you editing? What platform are you on?

**Comment by JonathanMagus** [comment]
No, this doesn't happen with any other typeface that I have checked against, either the ones that I have made and am working on (like this one), or random samples of fonts made by others.

It is a persistent thing though. So, if Fontforge is closed and the system shut down and restarted the greyed out header is still there on reopening.

I am using Windows 7 Professional. I could cross-check this on a Mac though to see if it shows the same in Fontforge on that.

The typeface is one that was in use around 1800 that is likely to be either by Caslon or Alexander Wilson. It is an early transitional, or step towards that style. Because the digital re-mastering imitates the original the outlines are comparatively complex (not just simple straight lines and curves) to reflect the roughness of the printing and ink gain around the characters. However, there are no self-intersects, or any errors in fact. It validates completely. The font also uses OpenType features.

There is no hinting because this kind of 'textured' outline is difficult to hint and produces overlapping hints if Auto-hinted. Plus the presentation gains nothing from it. So it has Blue Values only. Since I haven't found a way of setting up 'Don't auto-hint' as a Fontforge default, at some stage each of the characters will have gone through a process of Auto-hinting, followed by the clearing of hints and then re-setting as 'Don't auto-hint', before saving. Thinking back the grey header possibly appeared after doing this for the 'quoteright', which was a late addition to the font. That said the same process has been followed with other typefaces in progress and nothing like this has happened elsewhere.

If the generated .otf file is reopened in Fontforge the grey header doesn't show as grey. It only occurs in the original Fontforge file.

As I said to begin with, I am not sure what this is, or what it means (if anything). It doesn't seem to have affected the outcome and there doesn't appear to be a problem with the font. It is the fact that it is always there that makes me think that Fontforge is unhappy about something. I have reported this as an observation of odd behaviour happening. Whether it is a bug or not I cannot say.

**Comment by JonathanMagus** [comment]
I have managed to fix this now and have worked out what it probably was. 

Whilst working on a completely different font I realised that Fontforge shows the grey header when there is something present on the background layer when the .sfd file is saved. Normally that is very evident because it has been placed there deliberately, to trace around, or for comparison with the foreground layer; and is either a black and white bitmap, or a clearly differentiated outline. (If an exact copy of the foreground outline is also on the background it would be less obvious though.) When the background element is removed and the file saved again the cell header reverts to white.

In this instance nothing had been placed on the background intentionally, so I am not sure how it got there, or indeed what it was, since on examining the background layer of the apostrophe (8217, quoteright) glyph in this particular font there was nothing to be seen; and not an exact replica of the foreground outline present, which is what I thought might have been the cause. So there was apparently nothing there at all, however, selecting 'Clear Background' from the 'Edit' tab removed something and on saving again the cell header was no longer greyed out. Whatever it was must have been well outside the 'canvas' of the foreground space.

I presume that this feature wasn't being transferred to the generated font and didn't appear as an error, because the background layer is not part of the finalised output.

That would appear to be the explanation. This was not a bug. It was Fontforge indicating that something was present that should not have been.
